src/pybind/cephfs.py

```diff
@@ -299,6 +307,40 @@ def chdir(self, path):
         if ret < 0:
             raise make_ex(ret, "chdir failed")
 
+    def isdir(self, dirent):
+        return dirent['d_type'] == 0x4
+
+    def opendir(self, path):
+        self.require_state("mounted")
+        if not isinstance(path, str):
+            raise TypeError('path must be a string')
+        dir_handler = c_void_p()
+        ret = self.libcephfs.ceph_opendir(self.cluster, c_char_p(path),
+                                          pointer(dir_handler));
+        if ret < 0:
+            raise make_ex(ret, "opendir failed")
+        return dir_handler
+
+    def readdir(self, dir_handler):
+        self.require_state("mounted")
+        self.libcephfs.ceph_readdir.restype = POINTER(cephfs_dirent)
+        while True:
+            dirent = self.libcephfs.ceph_readdir(self.cluster, dir_handler)
+            if not dirent:
+                return None
+
+            return {'d_ino': dirent.contents.d_ino,
```

Comment:
It would be nice to return an object instead of a class here, so that the API consumer knows the fields are well defined.  Perhaps use a namedtuple() for speed?